runtime/bcverify/rtverify.c

```diff
@@ -1786,6 +1794,15 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
 				/* put a uninitialized object of the correct type on the stack */
 				PUSH(BCV_SPECIAL_NEW | (start << BCV_CLASS_INDEX_SHIFT));
 				break;
+			
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+			case JBdefaultvalue: /* TODO: ValueTypes perform actual verification */
+				index = PARAM_16(bcIndex, 1);
+				info = &constantPool[index];
+				utf8string = J9ROMSTRINGREF_UTF8DATA((J9ROMStringRef *) info);
+				stackTop = pushClassType(verifyData, utf8string, stackTop);
```

Comment:
Does the type being pushed here include the `Q`?  How does it differentiate from pushing the `L` version of the class?

Reply:
Normally, when instantiating a new `L` object, it will first be pushed (`JBnew`), then later initialized. `JBdefaultValue` should push a reference to a allocated instance of the value class which has been initialized to default values.

As was noted in #3731, `vrfyhelp.c`'s `pushClassType` method does not differentiate between Q and L types.

Reply:
Looks like `utf8string` does not contain `Q` or `;`, just the class name. However, `convertClassNameToStackMapType` which does the eventual conversion from string to type searches for the romClass matching the name. As a result, the Q or L type data should be retrievable at the point where it is required.